dask/layers.py

```diff
@@ -103,6 +105,165 @@ def __init__(
         )
 
 
+class SlicingLayer(Layer):
+    """Array slicing HighLevelGraph Layer"""
+
+    def __init__(
+        self,
+        out_name,
+        in_name,
+        blockdims,
+        new_blockdims,
+        index,
+        shape,
+        parts_out=None,
+    ):
+        super().__init__()
+        self.out_name = out_name
+        self.in_name = in_name
+        self.blockdims = blockdims
+        self.new_blockdims = new_blockdims
+        self.index = index
+        self.shape = shape
+        self.parts_out = parts_out
+
+    def __repr__(self):
+        return "SlicingLayer<name='{}'".format(self.out_name)
+
+    @property
+    def _dict(self):
+        """Materialize full dict representation"""
+        if hasattr(self, "_cached_dict"):
+            return self._cached_dict
+        else:
+            dsk = self._construct_graph()
+            self._cached_dict = dsk
+        return self._cached_dict
+
+    def __getitem__(self, key):
+        return self._dict[key]
+
+    def __iter__(self):
+        return iter(self._dict)
+
+    def __len__(self):
+        return len(self._dict)
+
+    def is_materialized(self):
+        return hasattr(self, "_cached_dict")
+
+    def get_output_keys(self):
+        return self.keys()  # FIXME! this implementation materializes the graph
+
+    def _get_block_slices(self, shape=None, blockdims=None, index=None):
+        """Get a list (for each dimension) of dicts{blocknum: slice()}"""
+        try:
+            block_slices = self._block_slices
+        except AttributeError:
+            block_slices = list(map(_slice_1d, shape, blockdims, index))
+            self._block_slices = block_slices
+        return block_slices
+
+    def _keys_to_output_parts(self, keys):
+        """Simple utility to convert keys to array chunk indices."""
+        parts = set()
```

Comment:
This may well be one of those instances of logic copied from the shuffle layer, and used in a context where it doesn't make sense anymore. What do you think?

Here's the original context, in `SimpleShuffleLayer`:
https://github.com/dask/dask/blob/a87236041ff223363698f65c5b7c153415a41259/dask/layers.py#L445-L456